epoxy-adapter/src/test/java/com/airbnb/epoxy/ViewTypeManagerTest.java

```diff
@@ -0,0 +1,113 @@
+package com.airbnb.epoxy;
+
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.FrameLayout;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RuntimeEnvironment;
+import org.robolectric.annotation.Config;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+
+@Config(sdk = 21, manifest = TestRunner.MANIFEST_PATH)
+@RunWith(TestRunner.class)
+public class ViewTypeManagerTest {
+
+  static class TestModel extends EpoxyModelWithView<View> {
+    @Override
+    protected View buildView(ViewGroup parent) {
+      return new FrameLayout(RuntimeEnvironment.application);
+    }
+  }
+
+  static class ModelWithViewType extends TestModel {
+
+    @Override
+    protected int getViewType() {
+      return 1;
+    }
+  }
+
+  static class ModelWithViewType2 extends TestModel {
+
+    @Override
+    protected int getViewType() {
+      return 2;
+    }
+  }
+
+  @Test
+  public void modelWithLayout() {
+    SimpleEpoxyAdapter adapter = new SimpleEpoxyAdapter();
+    adapter.addModel(new ModelWithViewType());
+    adapter.addModel(new ModelWithViewType2());
+
+    // The view type should be the value declared in the model
+    assertEquals(1, adapter.getItemViewType(0));
+    assertEquals(2, adapter.getItemViewType(1));
```

Comment:
Yeah, this isn't a perfect architecture in that the adapter and view type manager are tightly coupled, so it's easiest to test through the adapter. I could make the interface more generic, or mock out the adapter, to make this a true unit test of the view type manager, but I don't think that's worth the extra overhead even though that would be better form. These are more like integration tests of the adapter/viewtypemanager usage so we can be sure that it is used correctly in the adapter.